sanitize MIDI parser

slarticodefast opened the pull request:
## About the PR
Admin logs were failing to be saved sometimes, if they contained a message about a player playing a broken MIDI file.
This means admin logs from the last 3 weeks may be incomplete.

## Why / Balance
Fixes https://github.com/space-wizards/space-station-14/issues/38744

## Technical details
The MIDI parser reads the instrument and track names from the byte stream. These later show up in the admin logs.
But since they were never sanitized they might contain the `0x00` byte, which causes database errors when contained in a string.

~~To fix it we simply remove any control characters from the ASCII string.
I'm using linq, but if preferred I could try doing this via a regex instead. But this should work as well.~~

We sanitize on the server in case a client might send modified strings.

## Requirements
<!-- Confirm the following by placing an X in the brackets [X]: -->
- [x] I have read and am following the [Pull Request and Changelog Guidelines](https://docs.spacestation14.com/en/general-development/codebase-info/pull-request-guidelines.html).
- [x] I have added media to this PR or it does not require an ingame showcase.
<!-- You should understand that not following the above may get your PR closed at maintainer’s discretion -->

## Breaking changes
none

**Changelog**
ADMIN:
:cl:
- fix: Fixed admin logs not being saved to the databases when a player played certain MIDI files.

Simyon264 commented on Content.Client/Instruments/MidiParser/MidiParser.cs:
```diff
@@ -103,13 +104,16 @@ public static bool TryGetMidiTracks(
                         var text = Encoding.ASCII.GetString(metaData, 0, (int)metaLength);
+                        // remove all control characters
+                        // some of these ended up in the admin logs, causing database errors
+                        var sanitizedText = new string(text.Where(c => !char.IsControl(c)).ToArray());
```
I would use regex here and assign to `text` instead of using a new variable.

slarticodefast commented:
Changed it to use a for loop instead of linq.
Also moved the sanitation to the server, so that anything the client is sending will be cleaned up.
I restricted it to non-control ascii chars, because that is what can be read from the MIDI file.
